### PR TITLE
GEODE-5013 Replace org.json with Jackson in GfJsonObject

### PR DESCRIPTION
**Pull request by bschuchardt**
This removes org.json as a dependency in geode-core and associated test
modules.  GfJsonObject and GfJsonArray are ported to use the same basic
Jackson ObjectMapper setup as QueryResultFormatter so that we now have a
uniform mechanism for managing JSON documents for querying/gfsh.

The geode-web-api and geode-pulse modules still use the geode-json
module.  When/if they are ported to use GfJsonObject/Array we can delete
the geode-json module.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/build.gradle [hunk 1]
```diff
@@ -271,7 +271,6 @@ dependencies {
   compile('com.healthmarketscience.rmiio:rmiio')
 
   compile(project(':geode-common'))
-  compile(project(':geode-json'))
```
Comment (agingade) [inline geode-core/build.gradle]: Removing this line; will this impact building geode-web-api and geode-pulse?

File: geode-core/src/integrationTest/java/org/apache/geode/distributed/LocatorLauncherRemoteIntegrationTestCase.java [hunk 2]
```diff
@@ -199,6 +199,7 @@ protected LocatorLauncher awaitStart(final LocatorLauncher launcher) {
       try {
         assertThat(launcher.status().getStatus()).isEqualTo(Status.ONLINE);
       } catch (Exception e) {
+        e.printStackTrace();
```
Comment (agingade) [inline geode-core/src/integrationTest/java/org/apache/geode/distributed/LocatorLauncherRemoteIntegrationTestCase.java]: remove it?

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/json/AbstractJSONFormatter.java [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+    @Override
+    public void serialize(Collection value, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeStartObject();
+      _serialize(value, gen);
+      gen.writeEndObject();
+    }
+
+    void _serialize(Collection value, JsonGenerator gen) throws IOException {
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/json/AbstractJSONFormatter.java]: remove underscore from method name....

File: geode-core/build.gradle (hunk 1 above)
Comment (bschuchardt) [reply geode-core/build.gradle]: The geode-web-api includes geode-json as a dependency.  There are tickets for getting rid of it there and in geode-pulse.

**Pull request closed without merging**